docs/ibc/light-clients/wasm/overview.md

```diff
@@ -0,0 +1,28 @@
+<!--
+order: 1
+-->
+
+# `08-wasm`
+
+## Overview
+
+Learn about the `08-wasm` light client proxy module. {synopsis}
+
+### Context
+
+Traditionally, light clients used by ibc-go have been implemented in Go, and since ibc-go v7 (with the release of the 02-client refactor), they are [first-class Cosmos SDK modules](../../../architecture/adr-010-light-clients-as-sdk-modules.md). This means that upgrading existing light client implementations or adding support for new light clients is a multi-step, time-consuming process involving on-chain governance: it is necessary to modify the codebase of ibc-go (if the light client is part of its codebase), re-build chains' binaries, pass a governance proposal and validators upgrade their nodes.
+
+Another problem stemming from the above process is that if a chain wants to upgrade its own consensus, it will need to convince every chain connected to it to upgrade its light client in order to stay connected, going through the on-chain governance process mentioned before. Due to the time consuming process required to upgrade a light client, a chain with lots of connections needs to be disconnected from its counterparties for a non-negligible amount of time after upgrading its consensus, which can be very expensive in terms of time and effort.
```

Comment:
I'm not sure if this is entirely true:
> it will need to convince every chain connected to it to upgrade its light client in order to stay connected

The chain which modifies its consensus only needs to pass a gov proposal to store its new lightclient in the upgrade path and then relayers upgrade the counterparties permissionlessly, correct me if I'm wrong.